src/main/java/rx/exceptions/CompositeException.java

```diff
@@ -49,12 +49,19 @@
     public CompositeException(String messagePrefix, Collection<? extends Throwable> errors) {
         Set<Throwable> deDupedExceptions = new LinkedHashSet<Throwable>();
         List<Throwable> _exceptions = new ArrayList<Throwable>();
-        for (Throwable ex : errors) {
-            if (ex instanceof CompositeException) {
-                deDupedExceptions.addAll(((CompositeException) ex).getExceptions());
-            } else {
-                deDupedExceptions.add(ex);
+        if (errors != null) {
+            for (Throwable ex : errors) {
+                if (ex instanceof CompositeException) {
+                    deDupedExceptions.addAll(((CompositeException) ex).getExceptions());
+                } else 
+                if (ex != null) {
+                    deDupedExceptions.add(ex);
+                } else {
+                    deDupedExceptions.add(new NullPointerException());
```

Comment:
Like throwing exception from an exception? Since you only have the option to detect null here, throwing an NPE would have the same stacktrace as these new NPEs, but with the exception that they will further 'crash' the application. In my way, these inconsistencies get captured and propagated with the CompositeException itself.